migrate/20241108_resource_group.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+Sequel.migration do
+  change do
+    create_enum(:resource_group_type, %w[dedicated shared])
+
+    create_table(:resource_group) do
+      column :id, :uuid, primary_key: true
+      column :name, :text, null: false
+      column :allocation_state, :allocation_state, default: "unprepared", null: false
+      column :type, :resource_group_type, default: "dedicated", null: false
+      column :allowed_cpus, :text, null: false
+      column :cores, Integer, null: false
+      column :total_cpu_percent, Integer, null: false
+      column :used_cpu_percent, Integer, null: false
+      column :total_memory_1g, Integer, null: false
+      column :used_memory_1g, Integer, null: false
```

Comment:
This will be needed for the allocator. The VmHost will "give" memory to the slice and then subtract from it for each VM allocated in the slice. I will need this in subsequent PR and I just wanted to minimize the number of migrations.